fv/infrastructure/infra_k8s.go

```diff
@@ -338,6 +411,31 @@ func (kds *K8sDatastoreInfra) AddNode(felix *Felix, idx int, needBGP bool) {
 	}
 }
 
+func (kds *K8sDatastoreInfra) ensureNamespace(name string) {
+	if isSystemNamespace(name) {
+		return
+	}
+
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+	}
+	_, err := kds.K8sClient.CoreV1().Namespaces().Create(ns)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (kds *K8sDatastoreInfra) PodExist(ns, podName string) bool {
+	if pods, err := kds.K8sClient.CoreV1().Pods(ns).List(metav1.ListOptions{}); err == nil {
+		for _, pod := range pods.Items {
+			if pod.Name == podName {
+				return true
```

Comment:
How about using `ListOptions.fieldSelector = "metadata.name=<podName>"` ?

(Doc for fieldSelector is in preparation at https://github.com/kubernetes/website/pull/9174/files)

Reply:
This function is not needed anymore because the changes made to health_test.go:133.